system_tests/user_system_tests.go

```diff
@@ -0,0 +1,100 @@
+package system_tests
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	rabbithole "github.com/michaelklishin/rabbit-hole/v2"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	"k8s.io/apimachinery/pkg/types"
+
+	topologyv1alpha1 "github.com/rabbitmq/messaging-topology-operator/api/v1alpha1"
+)
+
+var _ = Describe("Users", func() {
+	var (
+		namespace = MustHaveEnv("NAMESPACE")
+		ctx       = context.Background()
+		user      *topologyv1alpha1.User
+	)
+
+	BeforeEach(func() {
+		user = &topologyv1alpha1.User{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-test",
+				Namespace: namespace,
+			},
+			Spec: topologyv1alpha1.UserSpec{
+				RabbitmqClusterReference: topologyv1alpha1.RabbitmqClusterReference{
+					Name:      rmq.Name,
+					Namespace: rmq.Namespace,
+				},
+				Name: "user-test",
+				Tags: []topologyv1alpha1.UserTag{"policymaker", "management"},
+			},
+		}
+	})
+
+	It("declares and deletes a user successfully", func() {
+		By("declaring user")
+		Expect(k8sClient.Create(ctx, user, &client.CreateOptions{})).To(Succeed())
+		var userInfo *rabbithole.UserInfo
+		Eventually(func() error {
+			var err error
+			userInfo, err = rabbitClient.GetUser(user.Spec.Name)
+			return err
+		}, 10, 2).Should(BeNil())
+
+		Expect(*userInfo).To(MatchFields(IgnoreExtras, Fields{
+			"Name":             Equal(user.Spec.Name),
+			"Tags":             Equal("policymaker,management"),
+			"HashingAlgorithm": Equal(rabbithole.HashingAlgorithmSHA512),
+		}))
+		Expect(userInfo.PasswordHash).NotTo(BeEmpty())
+
+		By("Creating a Secret with the generated credentials")
+		generatedSecretKey := types.NamespacedName{
+			Name:      "user-test-user-credentials",
+			Namespace: namespace,
+		}
+		var generatedSecret = &corev1.Secret{}
+		Expect(k8sClient.Get(ctx, generatedSecretKey, generatedSecret)).To(Succeed())
```

Comment:
You can assert that the created secret keys `username` and `password`. We are conforming to a specific service spec with these keys, and should have a test for them :)